src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs

```diff
@@ -1,20 +1,142 @@
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
 
 namespace Microsoft.VisualStudio.Telemetry
 {
     [Export(typeof(ITelemetryService))]
-    internal class VsTelemetryService : ITelemetryService
+    [Export(typeof(IVsTelemetryService))]
+    internal class VsTelemetryService : IVsTelemetryService, ITelemetryService
     {
+        private const string EventPrefix = "vs/projectsystem/managed/";
+        private const string PropertyPrefix = "VS.ProjectSystem.Managed.";
+        private readonly IUnconfiguredProjectCommonServices _unconfiguredProjectCommonServices;
+        private readonly object _lock =  new object();
+
+        private TelemetryEventCorrelation _telemetryEventCorrelation;
+        private bool _telemetryEventCorrelationInitialized;
+        private Guid _projectGuid;
+
+        [ImportingConstructor]
+        public VsTelemetryService(IUnconfiguredProjectCommonServices unconfiguredProjectCommonServices)
+        {
+            Requires.NotNull(unconfiguredProjectCommonServices, nameof(unconfiguredProjectCommonServices));
+
+            _unconfiguredProjectCommonServices = unconfiguredProjectCommonServices;
+        }
+
+        /// <summary>
+        /// Creates the Project Correlation asset which will be used to correlate all the telemetry events for this project
+        /// identified uniquely through the Project Guid.
+        /// </summary>
+        /// <returns></returns>
+        private async Task<bool> CreateProjectCorrelationAssetAsync()
+        {
+            var configurationGeneralProperties = await _unconfiguredProjectCommonServices
+                                                .ActiveConfiguredProjectProperties
+                                                .GetConfigurationGeneralPropertiesAsync()
+                                                .ConfigureAwait(false);
+
+            if (Guid.TryParse((string)await configurationGeneralProperties.ProjectGuid.GetValueAsync().ConfigureAwait(false), out Guid guid))
+            {
+                _projectGuid = guid;
+            }
+            else
+            {
+                return false;
+            }
+
+            var targetFrameworkProperty = await configurationGeneralProperties.TargetFramework.GetEvaluatedValueAtEndAsync().ConfigureAwait(false);
+            var targetFrameworksProperty = await configurationGeneralProperties.TargetFrameworks.GetEvaluatedValueAtEndAsync().ConfigureAwait(false);
+            var outputTypeProperty = await configurationGeneralProperties.OutputType.GetEvaluatedValueAtEndAsync().ConfigureAwait(false);
+
+            lock (_lock)
+            {
+                if (!_telemetryEventCorrelationInitialized)
+                {
+                    var eventName = "Project";
+                    var projectProperties = new Dictionary<string, object>();
+
+                    if (!string.IsNullOrEmpty(targetFrameworksProperty))
```

Comment:
Project loaded event, in this case, will only be useful, if this event is correlated with some root asset. Without correlation, that event is just orphaned data, which says __this__ project got loaded with __these__ properties at __some__ point in time by __some instance__.

If we go the approach in the PR, we will know the properties of the project during initialization and use the telemetry events to understand the timeline of the actions that were performed on this project.